### PR TITLE
Migrate `pallet-mmr` to the new pallet attribute macro

### PR DESCRIPTION
**Pull request by koushiro**
Part of #7882

Migrate the `pallet-mmr` to the new pallet attribute macro

diff mmr-before.json mmr-after.json (metadata of substrate/bin/node):

```
4c4
<     "prefix": "MerkleMountainRange",
---
>     "prefix": "Mmr",
```

## :warning:  Breaking Change :warning:

From https://crates.parity.io/frame_support/attr.pallet.html#checking-upgrade-guidelines

> storages now use PalletInfo for module_prefix instead of the one given to decl_storage: Thus any use of this pallet in construct_runtime! should be careful to update name in order not to break storage or to upgrade storage (moreover for instantiable pallet). If pallet is published, make sure to warn about this breaking change.

So users of the pallet must be careful about the name they used in `construct_runtime!`. Hence the `runtime-migration` label, which might not be needed depending on the configuration of the pallet.

`polkadot` / `kusama` / `westend` don't use `pallet-mmr` for runtime, so no need for migration there.
(However, `rococo` used `pallet-mmr`, I'm not sure if need to migrate storage for `rococo`)

/cc @shawntabrizi @thiolliere

### CODE REVIEW

File: frame/merkle-mountain-range/src/lib.rs [hunk 1]
```diff
@@ -74,86 +71,105 @@ mod mock;
...
-	type OnNewRoot: primitives::OnNewRoot<<Self as Config<I>>::Hash>;
+	/// This pallet's configuration trait
+	#[pallet::config]
+	pub trait Config<I: 'static = ()>: frame_system::Config {
+		/// Prefix for elements stored in the Off-chain DB via Indexing API.
+		///
+		/// Each node of the MMR is inserted both on-chain and off-chain via Indexing API.
+		/// The former does not store full leaf content, just it's compact version (hash),
+		/// and some of the inner mmr nodes might be pruned from on-chain storage.
+		/// The later will contain all the entries in their full form.
```
Comment (tomusdrw) [inline frame/merkle-mountain-range/src/lib.rs:98]:
```suggestion
		/// The latter will contain all the entries in their full form.
```

**Review by tomusdrw — APPROVED** [review]
Nice! This is only deployed on Rocooco/Wococo now so we don't necessarily need to take care of the migration, but we should restart Rococo when that lands.

**Comment by tomusdrw** [comment]
@koushiro would you be so kind to prepare a polkadot companion PR (migrating the existing instance)?

File: frame/merkle-mountain-range/src/benchmarking.rs [hunk 2]
```diff
@@ -30,15 +30,15 @@ benchmarks! {
 		let leaves = x as u64;
```
Comment (thiolliere) [inline frame/merkle-mountain-range/src/benchmarking.rs:30]: for instantiable pallets, you should use the macro `benchmarks_instance_pallet` above

Comment (thiolliere) [reply frame/merkle-mountain-range/src/benchmarking.rs:30]: I fixed it,

Note: it wasn't introduce by the PR, in old macro, ppl should use benchmarks_instance

**Review by thiolliere — APPROVED** [review]

**Comment by thiolliere** [comment]
@tomusdrw is it ok to merge once green ? or should I ping somebody about the rococo restart ?

**Comment by tomusdrw** [comment]
We were planning to do Rococo runtime upgrade/restart anyway, so IMHO good to merge.

**Comment by thiolliere** [comment]
bot merge

**Pull request merged**